src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+    private final DtoResponseMapper<OrderResponseDto, Order> orderDtoResponseMapper;
+
+    public OrderController(OrderService orderService, UserService userService,
+                           ShoppingCartService shoppingCartService,
+                           DtoResponseMapper<OrderResponseDto, Order> orderDtoResponseMapper) {
+        this.orderService = orderService;
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+        this.orderDtoResponseMapper = orderDtoResponseMapper;
+    }
+
+    @PostMapping("{/userId}")

```

Comment:
1. Incorrect usage of @PostMapping annotation. The path variable should not have a leading slash.
```suggestion
    @PostMapping("/{userId}")

```
2. Task require implementation of endpoints with RequestParam mostly, not PathVariable, please recheck all controllers for same issue